tasks/mrrc-collect-params/mrrc-collect-params.yaml

```diff
@@ -0,0 +1,60 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: mrrc-collect-params
+spec:
+  params:
+    - name: dataJsonPath
+      description: path to data json file
+  results:
+    - name: mrrcParamFilePath
+      description: path of the env file for following tasks
+    - name: cosignPubKeyFile
+      description: the public key file name for cosign verification
+    - name: cosignPubKeySecret
+      description: the secret name for cosign verify public key
+    - name: charonAWSSecret
+      description: the secret name for charon aws credential file
+  steps:
+    - name: mrrc-collect-params
+      image: quay.io/konflux-ci/yq:latest
+      script: |
+        set -eux
+
+        cd $(workspaces.data.path)
+        mrrcEnvFile="./mrrc.env"
+        if [ -f $mrrcEnvFile ]; then
+          rm $mrrcEnvFile
+        fi
+
+        zipRegistry=$(yq -r '.zipRegistry' $(params.dataJsonPath))
+        echo "export MRRC_ZIP_REGISTRY=$zipRegistry" >> $mrrcEnvFile
+
+        environment=$(yq -r '.mrrc.environment' $(params.dataJsonPath))
+        release=$(yq -r '.mrrc.release' $(params.dataJsonPath))
+        target=$environment-maven-$release
+        echo "export MRRC_TARGET=$target" >> $mrrcEnvFile
+
+        productName=$(yq -r '.mrrc.product.name' $(params.dataJsonPath))
+        productVersion=$(yq -r '.mrrc.product.version' $(params.dataJsonPath))
+        echo "export MRRC_PRODUCT_NAME=$productName" >> $mrrcEnvFile
+        echo "export MRRC_PRODUCT_VERSION=$productVersion" >> $mrrcEnvFile
+
+        awsSecret=$(yq -r '.mrrc.awsSecret' $(params.dataJsonPath))
+        echo -n "$awsSecret" > "$(results.charonAWSSecret.path)"
+
+        consignPubKey=$(yq -r '.cosignPubKeyFile' $(params.dataJsonPath))
+        if [ "$consignPubKey" != "" ] && [ "$consignPubKey" != "null" ]; then 
```

Comment:
If the key is missing, I would think it is (always) either the empty string or null, so we would only have to check that one condition, not both

Reply:
um... I'll check the exact value for the jq empty search and use it later.

Reply:
Seems the jq will return "null" for the empty value, so will just use the null check. Fixed.

Reply:
OK, but I wonder what the intention is here. If it's not set, we will not save this result. I'm not sure right now, but won't it be a problem when you try to use that result value as input for the other task? Wouldn't it be better to just default to ""?

And what about all the other values this script loads? Only this one is checked somehow, but for example if `.mrrc.environment` and `.mrrc.release` are not set, you will end up with target=null-maven-null

Reply:
If it is a required value, you can add `-e` to the `jq` command so the task will fail if it is missing. We use this elsewhere

Reply:
Actually these values will set in RPA later for non-missing values, but of course, we can add -e to make sure the value will not be missing. I'll add them.

Reply:
If they are needed, as in whatever task(s) will use the result and will fall over if the value is the empty string, we should add the `-e`

Reply:
Added in fix.